tests/utils.py

```diff
@@ -0,0 +1,304 @@
+			# grab the highest value in the column
+			if float(row[col]) > value:
+				value = float(row[col])
+				epoch = row[0]
+	return value, epoch
+
+
+def getPredictedVal(filepath, metric):
+	"""Gets the relevant value from the trained model output
+
+	Parameters:
+	filepath (str): path to the model output file
+	metric (str): test metric, eg 'pearson' or 'f1_prod'
+
+	Returns:
+	float: Value to run test against
+	"""
+
+	with open(filepath) as file:
+		reader = csv.reader(file)
+		rownum = 0
+		row = next(reader)
+		col = row.index(metric)  # in header find column corresponding to metric
+		for row in reader:
+			# grab the latest value in the column
+			if reader.line_num > rownum:
+				value = row[col]
+				rownum = reader.line_num
+	return float(value)
+
+
+def getTestData_BiRNN_sent():
+    # Download and extract the files
+    def downloadAndExtractFiles(cachePath,*args):
+        for url in args:
+            fileName = os.path.basename(os.path.normpath(url))
+            checkPath = cachePath +  fileName
+            if os.path.exists(checkPath):
+                print(checkPath + ' already downloaded')
+            else :
+                print('\n Downloading ' + fileName + ' from ' + 'url')
+                download_to = os.path.join(cachePath, fileName)
+                with urllib.request.urlopen(url) as f:
+                    with open(download_to, "wb") as out:
+                        out.write(f.read())
+        print('\n')
+
+    baseCacheDir = 'cache/'
+    task = 'testData-sent/'
+    rawtask = 'raw-'+task
+
+    train = 'http://www.quest.dcs.shef.ac.uk/wmt15_files/task1_en-es_training.tar.gz'
+    dev = 'http://www.quest.dcs.shef.ac.uk/wmt15_files/task1_en-es_dev.tar.gz'
+    test = 'http://www.quest.dcs.shef.ac.uk/wmt15_files/task1_en-es_test.tar.gz'
+    label = 'http://www.quest.dcs.shef.ac.uk/wmt15_files/gold/Task1_gold.tar.gz'
+
+    os.makedirs(baseCacheDir, exist_ok=True)
+    cachePath = baseCacheDir + rawtask
+    os.makedirs(cachePath, exist_ok=True)
+
+    downloadAndExtractFiles(cachePath,train,dev,test,label)
+
+    for file in os.listdir(cachePath):
+        if file.endswith(".tar.gz"):
+            tar = tarfile.open(cachePath+file, "r:gz")
+            print('Extracting: ' + file + ' to ' + cachePath)
+            tar.extractall(path=cachePath)
+            tar.close()
+
+    # make the test data
+    dataDir = 'data/'+task
+    os.makedirs(dataDir, exist_ok=True)
+
+    totalLines = 500 # total number of lines to take from example data
+    for f in os.listdir( cachePath ):
+        if f.endswith(".hter") or f.endswith(".pe") or f.endswith(".source") or f.endswith(".target"):
+            file_in = cachePath+f
+            file_out = dataDir+f
+            print('Copying first ' + str(totalLines) + ' lines of ' + file_in + ' to ' + file_out)
+            with open(file_in) as file:
+                lines = file.readlines()
+                lines = [l for i, l in enumerate(lines) if i <= totalLines-1]
```

Comment:
`i < totalLines` neater than `i <= totalLines-1`. You've done that elsewhere, so this is presumably me failing to spot this on some previous occasion.

Reply:
Yes, sorry, this must be a hangover from some previous foible.
Resolved by 421fb96